### PR TITLE
Added notifications plugin to 2.3 manifest files.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
Signed-off-by: AWSHurneyt <hurneyt@amazon.com>

### Description
Added notifications plugin to 2.3 manifest files.

### Issues Resolved
https://github.com/opensearch-project/notifications/issues/516

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AWSHurneyt** [comment]
I'll revise this PR once the 2.3 branch is cut for the notifications plugin.

**Comment by AWSHurneyt** [comment]
> Hey @AWSHurneyt can you please resolve the conflict as well.

Will do. I'll fix the merge conflict when I adjust the `ref` to the 2.3 branch.

**Comment by gaiksaya** [comment]
@AWSHurneyt Pleas fix merge-conflicts too. Thanks!

File: manifests/2.3.0/opensearch-2.3.0.yml [hunk 1]
```diff
@@ -91,3 +91,21 @@ components:
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
+- name: notifications-core
```
Comment (gaiksaya) [inline manifests/2.3.0/opensearch-2.3.0.yml:94]: Need to fix the Indentation error

**Review by gaiksaya — CHANGES_REQUESTED** [review]

**Comment by AWSHurneyt** [comment]
Closing in favor of https://github.com/opensearch-project/opensearch-build/pull/2581

**Pull request closed without merging**